### PR TITLE
fix: add camera permission check to prevent black screen when scanner opened

### PR DESCRIPTION
**Pull request by bryce-mcmath**
# Summary of Changes

Added a unified camera permission check when scan is pressed. Previously, if a user selected the native camera permission modal option "Deny" or "Don't allow" they would be met with a black screen. Additionally, if they had already denied camera permission and pressed scan they would be again met with a black screen.

Here is the flow when a user presses scan for the first time:
permission_allow

Here is the flow when a user presses scan and has already given permission:
already_allowed

Here is the flow when a user presses scan and has already denied permission:
already_denied

Things to note:

Permission statuses: there are three main statuses for permissions; GRANTED, BLOCKED, and DENIED. DENIED means they have not granted permissions, but doesn't specify if they have actively refused the permission request or just haven't been asked yet. BLOCKED means they have been asked and have denied the request, but this status is only available through using the `request` method from `react-native-permissions`, rather than the `check` method. Using the `request` method will pop up the native permission modal if it hasn't already been BLOCKED. Hopefully that provides a bit of explanation for why the flow works the way it does.

Privacy state: the new CameraDisclosureModal replaces the old privacy policy popup screen (CameraDisclosure). Because of this, there is no need and no use of `didShowCameraDisclosure` state, so I've cut that out.

DismissiblePopupModal: the new flow required a dismissible popup modal, which I couldn't find an already-built component for, so I made this component which is only somewhat generalized but can easily be extended in the future if need be. It can be dismissed if the user taps outside of the modal or if they tap the clear icon. Its styles were adapted from a combination of the `InfoBox` and `PopupModal` components for consistency.

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: app/ios/Podfile [hunk 1]
```diff
@@ -22,6 +22,10 @@ end
 target 'AriesBifold' do
   config = use_native_modules!
 
+  permissions_path = '../node_modules/react-native-permissions/ios'
```
Comment (jleach) [inline app/ios/Podfile:25]: Were these two lines added automatically or manually?

File: core/App/components/modals/CameraDisclosureModal.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    },
+    buttonContainer: {
+      paddingTop: 10,
+    },
+  })
+
+  const onAllowTouched = async () => {
+    setRequestInProgress(true)
+    const granted = await requestCameraUse()
+    if (!granted) setShowSettingsPopup(true)
```
Comment (jleach) [inline core/App/components/modals/CameraDisclosureModal.tsx]: Can you use braces here? Its handy short hand but can lead to errors if someone less experienced picks up JS. One might be tempted to add a line below as part of the `if`.

File: core/App/components/modals/DismissiblePopupModal.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  const { t } = useTranslation()
+  const { TextTheme, ColorPallet } = useTheme()
+  const iconSize = 30
+
+  const styles = StyleSheet.create({
+    modalCenter: {
+      flex: 1,
+      justifyContent: 'center',
+      alignItems: 'center',
+      backgroundColor: 'rgba(0,0,0,0.5)',
```
Comment (jleach) [inline core/App/components/modals/DismissiblePopupModal.tsx]: Curious if this is in the ColorPallet?

File: core/App/screens/Scan.tsx [hunk 4]
```diff
@@ -54,17 +58,58 @@ const Scan: React.FC<ScanProps> = ({ navigation }) => {
     }
   }
 
-  const didDismissCameraDisclosure = () => {
-    dispatch({ type: DispatchAction.DID_SHOW_CAMERA_DISCLOSURE, payload: [] })
+  const permissionFlow = async (method: PermissionContract, permission: Permission) => {
```
Comment (jleach) [inline core/App/screens/Scan.tsx]: You have a return type below in `requestCameraUse`, best to add one here for consistency.

**Review by jleach — CHANGES_REQUESTED** [review]
Looks good. A few small comments.

File: app/ios/Podfile (hunk 1 above)
Comment (bryce-mcmath) [reply app/ios/Podfile:25]: Manually, following the installation instructions here: https://github.com/zoontek/react-native-permissions#setup

File: core/App/components/modals/CameraDisclosureModal.tsx (hunk 2 above)
Comment (bryce-mcmath) [reply core/App/components/modals/CameraDisclosureModal.tsx]: Will do

File: core/App/screens/Scan.tsx (hunk 4 above)
Comment (bryce-mcmath) [reply core/App/screens/Scan.tsx]: Will do

File: core/App/components/modals/DismissiblePopupModal.tsx (hunk 3 above)
Comment (bryce-mcmath) [reply core/App/components/modals/DismissiblePopupModal.tsx]: Doesn't seem to be. I'll add it, since it's used in a few places now

**Review by jleach — APPROVED** [review]

**Pull request merged**